### PR TITLE
Refactor: Replace & Update Custom Element Mixin 

### PR DESCRIPTION
**Pull request by sghoweri**
I noticed when working on a separate issue that the default custom element styling / margin bottom styles for several components was missing (the block-list, ordered-list, and unordered-list to be specific).

Due to changes in how our code is compiled and distributed downstream, the original solution @mikemai2awesome and I came up with no longer works so I iterated to fix the issue itself + add a few new options discussed (including the ability to add default margin-bottom spacing + optionally embed normalize resets in the component itself).

### CODE REVIEW

**Comment by sghoweri** [comment]
@mikemai2awesome I realize the logic in this new mixin is (intentionally) simple but I figure since this isn't getting used in a ton of places we could start small and build this up as needed.

File: packages/core/styles/02-tools/tools-custom-element/_tools-custom-element.scss [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  $elementName,
+  $defaultDisplay: inline,
+  $includeDefaultSpacing: null,
+  $includeNormalize: false
+){
+  #{$elementName}{
+    display: $defaultDisplay;
+
+
+    @if $includeDefaultSpacing == true or
```
Comment (mikemai2awesome) [inline packages/core/styles/02-tools/tools-custom-element/_tools-custom-element.scss]: `includeDefaultSpacing` would mean different things based on the display. If block, it's `margin-block-end: medium;`, if inline, it's `margin-inline-end: xsmall`. Also, should this be a true/false boolean?

File: packages/core/styles/02-tools/tools-custom-element/_tools-custom-element.scss [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+@function bolt-list-contains($list, $value) {
+  @return not not index($list, $value);
+}
+
+
+// helper / setup mixin when defining new custom elements.
+// optionally, can auto-include default spacing + normalize in the new custom element.
+@mixin bolt-custom-element(
+  $elementName,
+  $defaultDisplay: inline,
```
Comment (mikemai2awesome) [inline packages/core/styles/02-tools/tools-custom-element/_tools-custom-element.scss]: Should this just be named `display`?

File: packages/core/styles/02-tools/tools-custom-element/_tools-custom-element.scss [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  $includeDefaultSpacing: null,
```
Comment (mikemai2awesome) [inline packages/core/styles/02-tools/tools-custom-element/_tools-custom-element.scss]: Should this just be named `spacing`?

File: packages/core/styles/02-tools/tools-custom-element/_tools-custom-element.scss [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+// @todo: move to separate function
+@function bolt-list-contains($list, $value) {
+  @return not not index($list, $value);
+}
+
+
+// helper / setup mixin when defining new custom elements.
+// optionally, can auto-include default spacing + normalize in the new custom element.
+@mixin bolt-custom-element(
```
Comment (mikemai2awesome) [inline packages/core/styles/02-tools/tools-custom-element/_tools-custom-element.scss]: Should this be named `bolt-custom-element-display`?

**Comment by sghoweri** [comment]
@mikemai2awesome looks great - thanks for the help getting this cleaned up!

**Pull request merged**